text/0000-add-pack-interact.md

```diff
@@ -0,0 +1,83 @@
+1. **Updates.** One thing Dockerfiles can't do is update hardcoded dependencies. We should be able to show when buildpacks, lifecycle, stack, etc, can be upgraded so that the user gets this sense of value.
+
+---
+
+This RFC acts as a gateway to subsequent functionality. Follow-up RFCs could include:
+
+1. **CVE detection.** A user should be able to view CVEs that a **(r) rebase** will remove.
+1. **Build Hijack.** A user should be able to open a shell for debugging, into a failed build container.
+1. **Edit Build Plan.** A user should be able to modify a build plan, before building.  
+1. **Export Build Plan.** A user should be able to "save" a modified build plan, for reproducibility.
+1. **View Layer Details.** A user should be able to view a filesystem diff by a plan "step" à la [dive](https://github.com/wagoodman/dive).
+1. **Revert Plan.** A user should be able to revert to last successful build plan. 
+1. **Even More.**
+
+# How it Works
+[how-it-works]: #how-it-works
+
+One can make extensive use of [rivo/tview](github.com/rivo/tview) to provide a UI, and handle user events, on the command-line. In Golang.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- None
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Find other means to entice the casual application developer into the buildpacks ecosystem.
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [Mockup](https://storage.googleapis.com/artifacts.cf-garden-windows-dev.appspot.com/pack-interact-edit.mp4)
+- [Mockup Src](https://github.com/aemengo/pack-visualize)
+- [Hijack Issue](https://github.com/buildpacks/pack/issues/62)
+- [Layer Metadata Issue](https://github.com/buildpacks/lifecycle/issues/411)
+
+# Issues to Create
+
+- buildpacks/pack: `pack interact` prompts you to choose from suggested builders [epic/pack-interact]
+- buildpacks/pack: `pack interact` prompts you to choose a build tag [good first issue][epic/pack-interact]
+- buildpacks/pack: `pack interact` conducts detect phase and shows interactive panel [epic/pack-interact]
+- buildpacks/pack: `pack interact` shows **image:**, **builder:** info on interactive panel [good first issue][epic/pack-interact]
+- buildpacks/pack: `pack interact` shows **plan:** info on interactive panel [epic/pack-interact]
+- buildpacks/pack: `pack interact` shows when updates are available [epic/pack-interact]
+- buildpacks/pack: `pack interact` allows you to **(u) update** [epic/pack-interact]
+- buildpacks/pack: `pack interact` allows you to **(b) build** [epic/pack-interact]
+- buildpacks/pack: `pack interact` allows you to **(e) edit** build plan before build [status/needs-discussion][epic/pack-interact]
+- buildpacks/pack: `pack interact` allows and prompts you to **(r) rebase** [epic/pack-interact]
```

Comment:
Can we get some clarification on "update" vs "rebase"?

Reply:
**Rebase** meaning when you update the `run:` image of your application
**Update** meaning when you update anything else (lifecycle, buildpack, etc)

I want to draw the distinction for the user because you would do 1 or the other for different reasons. You would _rebase_ for security reasons. And you _update_ primarily to opt-in to potentially new functionality. Furthermore, by drawing the distinction, it bumps the "number special things that buildpacks can do".

Reply:
But isn't an update the same as a rebuild (aka build). I'm still confused as to why there would be a distinction. From pack's perspective we only have two operations. `build` and `rebase`. An update as you describe it is simply a `build` that points to a pre-existing image name as the output. Are you proposing the introduction of a new operation?

Reply:
> An update as you describe it is simply a build that points to a pre-existing image name as the output.

Yes

> Are you proposing the introduction of a new operation?

Yes. But I admit that I have not made the parameters very clear here. Much like a language package manager can lock dependencies at a specific version, I think it's a good pattern to follow. And these specifc versions can be bumped w/ **(u) update**.

Here's an example:
```
paketo-buildpacks/go-dist:0.3.0
```

changes to 

```
paketo-buildpacks/go-dist:0.3.1
```